src/main/java/org/folio/services/domainevent/CommonDomainEventPublisher.java

```diff
@@ -149,7 +149,7 @@ public <R> Future<Long> publishStream(ReadStream<R> readStream,
       });
 
     return promise.future()
-      .onComplete(e -> kafkaProducer.close())
+      .onComplete(e -> kafkaProducer.flush(x -> kafkaProducer.close()))
```

Comment:
```suggestion
      .eventually(e -> kafkaProducer.flush())
      .eventually(e -> kafkaProducer.close())
```

Reply:
According to my experiments, it is sufficient to wait for the close to happen.

```suggestion
      .eventually(x -> kafkaProducer.close())
```

There is a similar problem in publish where onComplete is used. That can be simplified too and actually wait for close to succeed or fail.